app_dart/test/service/luci_build_service_test.dart

```diff
@@ -0,0 +1,316 @@
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      final bool result = await service.scheduleBuilds(
+        prNumber: 1,
+        commitSha: 'abc',
+        repositoryName: 'cocoon',
+      );
+      expect(result, isFalse);
+    });
+    test('emptyListBuild', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[],
+        );
+      });
+      config.luciTryBuildersValue =
+          (json.decode('[{"name": "Cocoon", "repo": "cocoon"}]')
+                  as List<dynamic>)
+              .cast<Map<String, dynamic>>();
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      final bool result = await service.scheduleBuilds(
+        prNumber: 1,
+        commitSha: 'abc',
+        repositoryName: 'cocoon',
+      );
+      expect(result, isTrue);
+    });
+    test('unsupportedRepo', () async {
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      expect(
+          () async => await service.scheduleBuilds(
+                prNumber: 1,
+                commitSha: 'abc',
+                repositoryName: 'notsupported',
+              ),
+          throwsA(const TypeMatcher<BadRequestException>()));
+    });
+  });
+
+  group('cancelBuilds', () {
+    setUp(() {
+      serviceAccountInfo = const ServiceAccountInfo(email: 'abc@abcd.com');
+      config = FakeConfig(deviceLabServiceAccountValue: serviceAccountInfo);
+      mockBuildBucketClient = MockBuildBucketClient();
+    });
+    test('emptyList', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[],
+        );
+      });
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      await service.cancelBuilds('cocoon', 1, 'abc', 'new builds');
+      verify(mockBuildBucketClient.batch(any)).called(1);
+    });
+    test('withScheduledBuild', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[
+            Response(
+                searchBuilds: SearchBuildsResponse(builds: <Build>[
+              Build(
+                id: 998,
+                builderId: BuilderId(
+                  project: 'flutter',
+                  bucket: 'prod',
+                  builder: 'Linux',
+                ),
+                status: Status.started,
+              )
+            ]))
+          ],
+        );
+      });
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      await service.cancelBuilds('cocoon', 1, 'abc', 'new builds');
+      verify(mockBuildBucketClient.batch(any)).called(2);
```

Comment:
Can we check the content of the cancel request instead of just a call count?

Reply:
Done